### PR TITLE
Migrate tests to sound mode, fix type error in MemoryDirectory

### PR DESCRIPTION
**Pull request by jonahwilliams**

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Lets try this again!

**Comment by goderbauer** [comment]
Looks like AppVeyor is not happy?

**Comment by jonahwilliams** [comment]
These tests are already failing, and I can't fix them since it is a dart:io issue

File: packages/file/pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: file
-version: 6.0.0
+version: 6.1.0
```
Comment (goderbauer) [inline packages/file/pubspec.yaml:2]: There's a version mismatch: The changelog said this would be 6.0.1

**Comment by goderbauer** [comment]
> These tests are already failing, and I can't fix them since it is a dart:io issue

Is there a dart bug filed for that?

File: packages/file/pubspec.yaml (hunk 1 above)
Comment (jonahwilliams) [reply packages/file/pubspec.yaml:2]: oops, good catch

**Comment by jonahwilliams** [comment]
https://github.com/google/file.dart/issues/139

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**